sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala

```diff
@@ -64,23 +66,26 @@ trait StreamingQuery {
 
   /**
    * Returns the current status of the query.
+   *
    * @since 2.0.2
    */
   def status: StreamingQueryStatus
 
   /**
-   * Returns current status of all the sources.
-   * @since 2.0.0
+   * Returns an array of the most recent [[StreamingQueryProgress]] updates for this query.
+   * The number of records retained for each stream is configured by
+   * `spark.sql.streaming.numProgressRecords`.
+   *
+   *  @since 2.1.0
    */
-  @deprecated("use status.sourceStatuses", "2.0.2")
-  def sourceStatuses: Array[SourceStatus]
+  def recentProgress: Array[StreamingQueryProgress]
```

Comment:
Hmmm, yeah maybe.  Its not clear to me that `progress` is inherently singular and `progresses` is kind of a mouthful.  It is maybe nice for `Array`s to always be plural though.